nixos/modules/config/networking.nix

```diff
@@ -187,11 +199,11 @@ in
         "rpc".source = pkgs.glibc.out + "/etc/rpc";
 
         # /etc/hosts: Hostname-to-IP mappings.
-        "hosts".text =
+        "hosts".text = let foo = concatStringsSep " " cfg.extraLocalHosts; in
           ''
-            127.0.0.1 localhost
+            127.0.0.1 localhost ${foo}
```

Comment:
As for prepending `extraHosts` before localhost, this will cause havoc, as explained in chosen answer of article.